File: Scripts/micro_dao/subtipo_servico.py
from micro_dao.objeto_modelo import ObjetoModelo

class SubtipoServico(ObjetoModelo):

    _id: int
    _nome: str
    _fk_tipo_servico: int

    def __init__(self, nome_tipo: str, fk_tipo_servico: int):
        self._id = 1
        self.set_nome(nome_tipo)
        self.set_fk_tipo_servico(fk_tipo_servico)

    def get_id(self) -> int:
        return self._id

    def set_id(self, id: int):
        self._id = id
        return self

    def get_nome(self) -> str:
        return self._nome

    def set_nome(self, nome_tipo: str):
        self._nome = nome_tipo
        return self

    def get_fk_tipo_servico(self) -> int:
        return self._fk_tipo_servico

    def set_fk_tipo_servico(self, fk_tipo_servico: int):
        self._fk_tipo_servico = fk_tipo_servico
        return self

File: Projeto_Java/src/Dominio/Entidades/SubtipoServico.java
package Dominio.Entidades;

import javax.persistence.*;
import java.util.ArrayList;
import java.util.Collection;

/**
 * Classe criada com a finalidade de permitir que o Hibernate tenha acesso a
 * todos subtipos de um serviço.
 */
@Entity
@Table(name = "subtipo_servico")
public class SubtipoServico {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id")
    private int id;

    @Column(name = "nome", length = 50)
    private String nome;

    @Column(name = "fk_tipo_servico")
    private int fkTipoServico;

    @ManyToMany(mappedBy = "subtiposServico")
    private Collection<Servico> servicos = new ArrayList<>();


    public SubtipoServico() {}

    // Apenas Getters, devido a classe se comportar como Enum;
    public String getNome() {
        return nome;
    }

    public int getFkTipoServico() {
        return fkTipoServico;
    }

    public int getId() {
        return id;
    }
}
